No commands working after 1.6.0 when P4-related environment variables are not set

Hey,

I am using atom and perforce on a daily basis. After upgrade to 1.6.0/1.6.1 it totally stopped working on my mac.

I have .p4config with p4password and i have .bashrc and .bash_profile. Nothing helped, i kept getting P4PASSWD not set, even if it was. Downgrading to 1.5.3 solved the problem.

What info can i provide so this get fixed ?

Regards,

Still no luck with that, upgrading to 1.6.2 didn't solve the problem.
I am on mac. Here's the console log : 

atom-perforce: loading environment vars via command "/bin/bash -c 'source ~/.profile >/dev/null && env'"
/Users/nbochev/.atom/packages/atom-perforce/lib/environment.js:69 atom-perforce: loading environment vars via command "/bin/bash -c 'source ~/.bash_profile >/dev/null && env'"
/Users/nbochev/.atom/packages/atom-perforce/lib/environment.js:69 atom-perforce: loading environment vars via command "/bin/bash -c 'source ~/.bashrc >/dev/null && env'"
/Applications/Atom.app/Contents/Resources/app.asar/src/deserializer-manager.js:42 No deserializer found for Object
index.js:46 Window load time: 2004ms
/Users/nbochev/.atom/packages/sync-settings/lib/sync-settings.coffee:60 checking latest backup...
/Users/nbochev/.atom/packages/sync-settings/lib/sync-settings.coffee:299 Creating GitHubApi client with token = XXXXXXXXXXXX
/Users/nbochev/.atom/packages/sync-settings/lib/tracker.coffee:71 tracking Activate
/Users/nbochev/.atom/packages/sync-settings/lib/sync-settings.coffee:66 null Object
/Users/nbochev/.atom/packages/sync-settings/lib/sync-settings.coffee:81 latest backup version xxxxxxxxxxxxxxxx
/Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:590 Error: Command failed: /bin/sh -c p4 -ztag opened  ./...
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Socket.<anonymous> (internal/child_process.js:319:11)
  at emitOne (events.js:77:13)
  at Socket.emit (events.js:169:7)
  at Pipe._onclose (net.js:467:12)
(anonymous function) @ /Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:590
/Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:590 Error: Command failed: /bin/sh -c p4 -ztag opened  ./...
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Socket.<anonymous> (internal/child_process.js:319:11)
  at emitOne (events.js:77:13)
  at Socket.emit (events.js:169:7)
  at Pipe._onclose (net.js:467:12)
(anonymous function) @ /Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:590
/Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:446 Error: Command failed: /bin/sh -c p4 diff  master.pp
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
(anonymous function) @ /Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:446
index.js:13 Unhandled promise rejection Object with error: Error: Command failed: /bin/sh -c p4 diff  master.pp
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
(anonymous function) @ index.js:13
/Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:714 Error: Command failed: /bin/sh -c p4 fstat  master.pp
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Socket.<anonymous> (internal/child_process.js:319:11)
  at emitOne (events.js:77:13)
  at Socket.emit (events.js:169:7)
  at Pipe._onclose (net.js:467:12)

index.js:13 Unhandled promise rejection Object with error: Error: Command failed: /bin/sh -c p4 fstat  master.pp
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Socket.<anonymous> (internal/child_process.js:319:11)
  at emitOne (events.js:77:13)
  at Socket.emit (events.js:169:7)
  at Pipe._onclose (net.js:467:12)
(anonymous function) @ index.js:13
/Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:590 Error: Command failed: /bin/sh -c p4 -ztag opened  ./...
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
(anonymous function) @ /Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:590
/Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:329 could not p4 sync Error: Command failed: /bin/sh -c p4 sync  ./...
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
(anonymous function) @ /Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:329
index.js:13 Unhandled promise rejection Object with error: "Command failed: /bin/sh -c p4 sync  ./...
Perforce password (P4PASSWD) invalid or unset.
"(anonymous function) @ index.js:13
/Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:714 Error: Command failed: /bin/sh -c p4 fstat  .p4config
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

index.js:13 Unhandled promise rejection Object with error: Error: Command failed: /bin/sh -c p4 fstat  .p4config
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
(anonymous function) @ index.js:13
/Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:446 Error: Command failed: /bin/sh -c p4 diff  .p4config
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Socket.<anonymous> (internal/child_process.js:319:11)
  at emitOne (events.js:77:13)
  at Socket.emit (events.js:169:7)
  at Pipe._onclose (net.js:467:12)
(anonymous function) @ /Users/nbochev/.atom/packages/atom-perforce/lib/atom-perforce.js:446
index.js:13 Unhandled promise rejection Object with error: Error: Command failed: /bin/sh -c p4 diff  .p4config
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

  at ChildProcess.exithandler (child_process.js:208:12)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:172:7)
  at maybeClose (internal/child_process.js:764:16)
  at Socket.<anonymous> (internal/child_process.js:319:11)
  at emitOne (events.js:77:13)
  at Socket.emit (events.js:169:7)
  at Pipe._onclose (net.js:467:12)

Here's the output of env:

$source ~/.bash_profile >/dev/null && env

TERM_PROGRAM=iTerm.app
SHELL=/bin/bash
TERM=xterm-256color
CLICOLOR=true
TMPDIR=/var/folders/yr/kt0jgk7d2l91ks5_vf926bvc002zn4/T/
DOCKER_HOST=tcp://192.168.99.100:2376
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.kPZhwGGEJV/Render
OLDPWD=/Users/nbochev
USER=nbochev
P4PASSWD=XXXXXXX
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.XsaP5iu9Ab/Listeners
__CF_USER_TEXT_ENCODING=0x17EA4:0x0:0x0
DOCKER_TLS_VERIFY=1
LSCOLORS=gxfxcxdxbxCgCdabagacad
PATH=/Library/PostgreSQL/9.4/bin:/usr/local/sbin:/usr/local/packer:/Library/Frameworks/Python.framework/Versions/3.5/bin:/Library/Frameworks/Python.framework/Versions/3.4/bin:/Users/nbochev/Development/go/bin:/build/apps/bin:/Library/Frameworks/Python.framework/Versions/2.7/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/aria2/bin:/usr/local/go/bin:/mts/git/bin
PWD=/Users/nbochev/Development/build
EDITOR=vi
LANG=en_US.UTF-8
ITERM_PROFILE=Default
XPC_FLAGS=0x0
PS1=[\033[38;5;11m]\u[$(tput sgr0)][\033[38;5;15m]@[$(tput sgr0)][\033[38;5;12m]\h[$(tput sgr0)][\033[38;5;15m]:[$(tput sgr0)][\033[38;5;6m][\w]:[$(tput sgr0)][\033[38;5;15m]\n\$[$(tput sgr0)]
XPC_SERVICE_NAME=0
DOCKER_CERT_PATH=/Users/nbochev/.docker/machine/machines/default
SHLVL=1
HOME=/Users/nbochev
LOGNAME=nbochev
GOPATH=/Users/nbochev/Development/go
P4CONFIG=.p4config
_=/usr/bin/env

i am starting atom by going to the project dir and typing :
$ atom .

Typing p4 info in that same dir, produces:

$p4 info
User name: nbochev
Client name: nbochev-build
Client host: wolfheart-MBPRO.local
Client root: /Users/nbochev/Development/build
Current directory: /Users/nbochev/Development/build/depot/build/devil
Peer address: xxxxxxx
Client address: xxxxxx
Server address: perforce-releng
Server root: /p4d/perforce-releng_1850
Server date: 2015/11/11 07:36:41 -0800 PST
Server uptime: 1270:07:38
Server version: P4D/LINUX26X86_64/2015.1/1217912 (2015/08/13)
ServerID: perforce-releng_1850
Proxy address: xxxxxxxxx
Proxy version: P4P/LINUX26X86_64/2015.1/1126382 (2015/06/22)
Server license: xxxxx
Server license-ip: xxxxxxx
Case Handling: sensitive

Running commands on the shell works as expected.

I just upgraded from 1.5.3 to 1.6.2 and am running into the same problem.  Interestingly, I don't store my password in a config file and normally use `p4 login` from the command line.  I've verified that I am logged in and commands are working from the command line. But inside Atom, I see the same errors as @sh1ny.

```
atom-perforce: loading environment vars via command "/bin/bash -c 'source ~/.profile >/dev/null && env'"
/Users/mdouglass/.atom/packages/atom-perforce/lib/environment.js:69 atom-perforce: loading environment vars via command "/bin/bash -c 'source ~/.bash_profile >/dev/null && env'"
/Users/mdouglass/.atom/packages/atom-perforce/lib/environment.js:69 atom-perforce: loading environment vars via command "/bin/bash -c 'source ~/.bashrc >/dev/null && env'"
index.js:46 Window load time: 1477ms
/Users/mdouglass/.atom/packages/atom-perforce/lib/environment.js:77 error executing command /bin/bash -c 'source ~/.profile >/dev/null && env': Error: Command failed: /bin/bash -c /bin/bash -c 'source ~/.profile >/dev/null && env'
/bin/bash: /Users/mdouglass/.profile: No such file or directory

/Users/mdouglass/.atom/packages/atom-perforce/lib/atom-perforce.js:446 Error: Command failed: /bin/sh -c p4 diff  index.js
Perforce password (P4PASSWD) invalid or unset.

Perforce password (P4PASSWD) invalid or unset.

    at ChildProcess.exithandler (child_process.js:208:12)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:764:16)
    at Socket.<anonymous> (internal/child_process.js:319:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:467:12)(anonymous function) @ /Users/mdouglass/.atom/packages/atom-perforce/lib/atom-perforce.js:446
```

--- a/lib/atom-perforce.js
+++ b/lib/atom-perforce.js
@@ -17,7 +17,8 @@ var atomPerforce = module, // sugary alias
         'P4PORT',
         'P4USER',
         'P4TICKETS',
-        'P4PASSWD'
+        'P4PASSWD',
+        'HOME'
     ],
     clientStatusBarTile,
     environmentReady;
@@ -704,7 +705,7 @@ atomPerforce.exports = {
 
         execP4Command('info', { cwd: dir })
         .then(function(p4Info) {
-            if (!p4Info['clientUnknown.'] && p4Info.currentDirectory.startsWith(p4Info.clientRoot)) {            
+            if (!p4Info['clientUnknown.'] && p4Info.currentDirectory.startsWith(p4Info.clientRoot)) {
                 return execP4Command('fstat', {
                     cwd: dir,
                     files: [path.basename(filepath)]

